school_library.rb

```diff
@@ -11,12 +13,44 @@ def initialize(age, name = 'Unknown', parent_permission: true)
 
   attr_accessor :age
 
-  Private def of_age
-    @age >= 18
-  end
   def can_use_services
     return unless @age >= 18 || parent_permission
 
     true
   end
+
+  def correct_name
+    @name
+  end
+
+  private
+
+  def of_age
+    @age >= 18
+  end
+end
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.upcase!
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.splice(0, 9)
+  end
 end
```

Comment:
- [ ] **[OPTIONAL]** Based on Ruby conventions, each class should be in its own file. I would recommend you move this class to a new file.